amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/RejectedTaskEvent.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package software.amazon.kinesis.coordinator;
+
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.kinesis.annotations.KinesisClientInternalApi;
+
+@Getter
+@ToString(exclude = "MESSAGE")
+@Slf4j
+@KinesisClientInternalApi
+class RejectedTaskEvent implements DiagnosticEvent {
+    private final String MESSAGE = "This is a misconfiguration of the thread pool. We currently " +
```

Comment:
This may not be due to misconfiguration. The MESSAGE can be "Review your thread configuration to prevent task rejections. Until next release, KCL will not be resilient to task rejections."

Reply:
Updated message